File: gamslib/lands/lands-scalar.py
#  LP written by GAMS Convert at 12/13/18 10:32:16
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         24        1       10       13        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         41       41        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        133      133        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x10 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x11 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x12 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x17 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x18 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x19 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x20 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x21 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x22 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x23 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x24 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x25 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x26 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x27 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x28 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x29 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x30 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x31 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x32 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x33 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x34 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x35 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x36 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x37 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x38 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x39 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x40 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x41 = Var(within=Reals,bounds=(0,None),initialize=0)

m.obj = Objective(expr=   10*m.x1 + 7*m.x2 + 16*m.x3 + 6*m.x4 + 12*m.x6 + 16*m.x7 + 12*m.x8 + 7.2*m.x9 + 9.6*m.x10
                        + 7.2*m.x11 + 1.2*m.x12 + 1.6*m.x13 + 1.2*m.x14 + 13.5*m.x15 + 18*m.x16 + 13.5*m.x17 + 8.1*m.x18
                        + 10.8*m.x19 + 8.1*m.x20 + 1.35*m.x21 + 1.8*m.x22 + 1.35*m.x23 + 9.6*m.x24 + 12.8*m.x25
                        + 9.6*m.x26 + 5.76*m.x27 + 7.68*m.x28 + 5.76*m.x29 + 0.96*m.x30 + 1.28*m.x31 + 0.96*m.x32
                        + 16.5*m.x33 + 22*m.x34 + 16.5*m.x35 + 9.9*m.x36 + 13.2*m.x37 + 9.9*m.x38 + 1.65*m.x39
                        + 2.2*m.x40 + 1.65*m.x41, sense=minimize)

m.c2 = Constraint(expr=   m.x1 + m.x2 + m.x3 + m.x4 >= 12)

m.c3 = Constraint(expr=   10*m.x1 + 7*m.x2 + 16*m.x3 + 6*m.x4 <= 120)

m.c4 = Constraint(expr= - m.x1 + m.x6 + m.x9 + m.x12 <= 0)

m.c5 = Constraint(expr= - m.x1 + m.x7 + m.x10 + m.x13 <= 0)

m.c6 = Constraint(expr= - m.x1 + m.x8 + m.x11 + m.x14 <= 0)

m.c7 = Constraint(expr= - m.x2 + m.x15 + m.x18 + m.x21 <= 0)

m.c8 = Constraint(expr= - m.x2 + m.x16 + m.x19 + m.x22 <= 0)

m.c9 = Constraint(expr= - m.x2 + m.x17 + m.x20 + m.x23 <= 0)

m.c10 = Constraint(expr= - m.x3 + m.x24 + m.x27 + m.x30 <= 0)

m.c11 = Constraint(expr= - m.x3 + m.x25 + m.x28 + m.x31 <= 0)

m.c12 = Constraint(expr= - m.x3 + m.x26 + m.x29 + m.x32 <= 0)

m.c13 = Constraint(expr= - m.x4 + m.x33 + m.x36 + m.x39 <= 0)

m.c14 = Constraint(expr= - m.x4 + m.x34 + m.x37 + m.x40 <= 0)

m.c15 = Constraint(expr= - m.x4 + m.x35 + m.x38 + m.x41 <= 0)

m.c16 = Constraint(expr=   m.x6 + m.x15 + m.x24 + m.x33 >= 3)

m.c17 = Constraint(expr=   m.x7 + m.x16 + m.x25 + m.x34 >= 5)

m.c18 = Constraint(expr=   m.x8 + m.x17 + m.x26 + m.x35 >= 7)

m.c19 = Constraint(expr=   m.x9 + m.x18 + m.x27 + m.x36 >= 3)

m.c20 = Constraint(expr=   m.x10 + m.x19 + m.x28 + m.x37 >= 3)

m.c21 = Constraint(expr=   m.x11 + m.x20 + m.x29 + m.x38 >= 3)

m.c22 = Constraint(expr=   m.x12 + m.x21 + m.x30 + m.x39 >= 2)

m.c23 = Constraint(expr=   m.x13 + m.x22 + m.x31 + m.x40 >= 2)

m.c24 = Constraint(expr=   m.x14 + m.x23 + m.x32 + m.x41 >= 2)

File: gamslib/lands/lands-scalar.jl
#  LP written by GAMS Convert at 10/11/20 12:39:51
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         24        1       10       13        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         41       41        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        133      133        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)
@variable(m, 0 <= x10, start=0)
@variable(m, 0 <= x11, start=0)
@variable(m, 0 <= x12, start=0)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= x17, start=0)
@variable(m, 0 <= x18, start=0)
@variable(m, 0 <= x19, start=0)
@variable(m, 0 <= x20, start=0)
@variable(m, 0 <= x21, start=0)
@variable(m, 0 <= x22, start=0)
@variable(m, 0 <= x23, start=0)
@variable(m, 0 <= x24, start=0)
@variable(m, 0 <= x25, start=0)
@variable(m, 0 <= x26, start=0)
@variable(m, 0 <= x27, start=0)
@variable(m, 0 <= x28, start=0)
@variable(m, 0 <= x29, start=0)
@variable(m, 0 <= x30, start=0)
@variable(m, 0 <= x31, start=0)
@variable(m, 0 <= x32, start=0)
@variable(m, 0 <= x33, start=0)
@variable(m, 0 <= x34, start=0)
@variable(m, 0 <= x35, start=0)
@variable(m, 0 <= x36, start=0)
@variable(m, 0 <= x37, start=0)
@variable(m, 0 <= x38, start=0)
@variable(m, 0 <= x39, start=0)
@variable(m, 0 <= x40, start=0)
@variable(m, 0 <= x41, start=0)

@objective(m, Min, 10*x1 + 7*x2 + 16*x3 + 6*x4 + 12*x6 + 16*x7 + 12*x8 + 7.2*x9 + 9.6*x10 + 7.2*x11 + 1.2*x12 + 1.6*x13
     + 1.2*x14 + 13.5*x15 + 18*x16 + 13.5*x17 + 8.1*x18 + 10.8*x19 + 8.1*x20 + 1.35*x21 + 1.8*x22 + 1.35*x23 + 9.6*x24
     + 12.8*x25 + 9.6*x26 + 5.76*x27 + 7.68*x28 + 5.76*x29 + 0.96*x30 + 1.28*x31 + 0.96*x32 + 16.5*x33 + 22*x34
     + 16.5*x35 + 9.9*x36 + 13.2*x37 + 9.9*x38 + 1.65*x39 + 2.2*x40 + 1.65*x41)

@constraint(m, x1 + x2 + x3 + x4 >= 12)

@constraint(m, 10*x1 + 7*x2 + 16*x3 + 6*x4 <= 120)

@constraint(m,  - x1 + x6 + x9 + x12 <= 0)

@constraint(m,  - x1 + x7 + x10 + x13 <= 0)

@constraint(m,  - x1 + x8 + x11 + x14 <= 0)

@constraint(m,  - x2 + x15 + x18 + x21 <= 0)

@constraint(m,  - x2 + x16 + x19 + x22 <= 0)

@constraint(m,  - x2 + x17 + x20 + x23 <= 0)

@constraint(m,  - x3 + x24 + x27 + x30 <= 0)

@constraint(m,  - x3 + x25 + x28 + x31 <= 0)

@constraint(m,  - x3 + x26 + x29 + x32 <= 0)

@constraint(m,  - x4 + x33 + x36 + x39 <= 0)

@constraint(m,  - x4 + x34 + x37 + x40 <= 0)

@constraint(m,  - x4 + x35 + x38 + x41 <= 0)

@constraint(m, x6 + x15 + x24 + x33 >= 3)

@constraint(m, x7 + x16 + x25 + x34 >= 5)

@constraint(m, x8 + x17 + x26 + x35 >= 7)

@constraint(m, x9 + x18 + x27 + x36 >= 3)

@constraint(m, x10 + x19 + x28 + x37 >= 3)

@constraint(m, x11 + x20 + x29 + x38 >= 3)

@constraint(m, x12 + x21 + x30 + x39 >= 2)

@constraint(m, x13 + x22 + x31 + x40 >= 2)

@constraint(m, x14 + x23 + x32 + x41 >= 2)
